src/dsymbol.c

```diff
@@ -978,6 +978,11 @@ Dsymbol *ScopeDsymbol::search(Loc loc, Identifier *ident, int flags)
                             a = mergeOverloadSet(a, s2);
                             continue;
                         }
+                        else if (ss->isModule() && s2->prot() == PROTprivate)
+                        {
+                            continue;
+                        }
```

Comment:
This is not bad, but does cover all cases?

1. `ss` is an overloadable public symbol, and
    a. `s2` is an overloadable public symbol.
    b. `s2` is an overloadable private symbol.
    c. `s2` is an non-overloadable public symbol.
    d. `s2` is an non-overloadable private symbol.

2. `ss` is an overloadable private symbol, and
    a. `s2` is an overloadable public symbol.
    b. `s2` is an overloadable private symbol.
    c. `s2` is an non-overloadable public symbol.
    d. `s2` is an non-overloadable private symbol.

3. `ss` is an non-overloadable public symbol, and
    a. `s2` is an overloadable public symbol.
    b. `s2` is an overloadable private symbol.
    c. `s2` is an non-overloadable public symbol.
    d. `s2` is an non-overloadable private symbol.

4. `ss` is an non-overloadable priate symbol, and
    a. `s2` is an overloadable public symbol.
    b. `s2` is an overloadable private symbol.
    c. `s2` is an non-overloadable public symbol.
    d. `s2` is an non-overloadable private symbol.